lib/utils/addEntries.js

```diff
@@ -2,123 +2,138 @@
- */
-function addEntries(compiler, options, server) {
-  // we're stubbing the app in this method as it's static and doesn't require
-  // a server to be supplied. createDomain requires an app with the
-  // address() signature.
-
-  const app = server || {
-    address() {
-      return { port: options.port };
-    },
-  };
-
-  /** @type {string} */
-  const domain = createDomain(options, app);
-  /** @type {string} */
-  const host =
-    options.client && options.client.host ? `&host=${options.client.host}` : '';
-  /** @type {string} */
-  let path = '';
-  // only add the path if it is not default
-  if (options.client && options.client.path && options.client.path !== '/ws') {
-    path = `&path=${options.client.path}`;
-  }
-  /** @type {string} */
-  const port =
-    options.client && options.client.port ? `&port=${options.client.port}` : '';
-  /** @type {string} */
-  const clientEntry = `${require.resolve(
-    '../../client/default/'
-  )}?${domain}${host}${path}${port}`;
-
-  /** @type {(string[] | string)} */
-  let hotEntry;
-
-  if (options.hot === 'only') {
-    hotEntry = require.resolve('webpack/hot/only-dev-server');
-  } else if (options.hot) {
-    hotEntry = require.resolve('webpack/hot/dev-server');
-  }
-  /**
-   * prependEntry Method for webpack 4
-   * @param {Entry} originalEntry
-   * @param {Entry} additionalEntries
-   * @returns {Entry}
-   */
-  const prependEntry = (originalEntry, additionalEntries) => {
-    if (typeof originalEntry === 'function') {
-      return () =>
-        Promise.resolve(originalEntry()).then((entry) =>
-          prependEntry(entry, additionalEntries)
-        );
-    }
-
-    if (typeof originalEntry === 'object' && !Array.isArray(originalEntry)) {
-      /** @type {Object<string,string>} */
-      const clone = {};
 
-      Object.keys(originalEntry).forEach((key) => {
-        // entry[key] should be a string here
-        const entryDescription = originalEntry[key];
-        clone[key] = prependEntry(entryDescription, additionalEntries);
-      });
+// @ts-ignore
+const EntryPlugin = webpack.EntryPlugin;
 
-      return clone;
-    }
-
-    // in this case, entry is a string or an array.
-    // make sure that we do not add duplicates.
-    /** @type {Entry} */
-    const entriesClone = additionalEntries.slice(0);
-    [].concat(originalEntry).forEach((newEntry) => {
-      if (!entriesClone.includes(newEntry)) {
-        entriesClone.push(newEntry);
-      }
-    });
-    return entriesClone;
-  };
+class DevServerPlugin {
+  /**
+   * @param {?Object} options - Dev-Server options
+   * @param {?Object} server
+   */
+  constructor(options, server) {
+    this.options = options;
+
+    // we're stubbing the app in this method as it's static and doesn't require
+    // a server to be supplied. createDomain requires an app with the
+    // address() signature.
+
+    this.app = server || {
+      address() {
+        return { port: options.port };
+      },
+    };
```

Comment:
Can you clarify? I am afraid about it

Reply:
I just copied from: https://github.com/webpack/webpack-dev-server/blob/6c2082fb8469af5025b024f7c2a7a5fa8fb03f53/lib/utils/addEntries.js#L20-L28 `git blame` shows it was added in https://github.com/webpack/webpack-dev-server/commit/e5dc236677e1865fce8afb903b9f833f7531333e.

Reply:
I think we should remove this and also remove this https://github.com/webpack/webpack-dev-server/blob/master/lib/Server.js#L1034, no need to do it anymore, if somebody will need this, they can use plugin